src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -1840,6 +1840,7 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
       info match {
         case ci @ ClassInfoType(_, _, _) =>
           setInfo(ci.copy(parents = ci.parents :+ SerializableTpe))
+          invalidateCaches(ci.typeSymbol.typeOfThis.underlying, ci.typeSymbol :: Nil)
```

Comment:
why `.underlying`? In the test case it doesn't matter (it reuturns the same instance). And looking at `invalidateCaches`, it seems counter-intuitive

```
  def invalidateCaches(t: Type, updatedSyms: List[Symbol]) =
    t match {
      case st: SingleType   if updatedSyms.contains(st.sym) => st.invalidateSingleTypeCaches()
      case tr: TypeRef      if updatedSyms.contains(tr.sym) => tr.invalidateTypeRefCaches()
      case ct: CompoundType if ct.baseClasses.exists(updatedSyms.contains) => ct.invalidatedCompoundTypeCaches()
      case _ =>
    }
``` 

you'd like to pass the `SingleType`, not its `underlying`.

Reply:
In the test case, removing `.underlying` does trigger a failure.

Adding:

```
println(showRaw(ci.typeSymbol.typeOfThis))
println(showRaw(ci.typeSymbol.typeOfThis.underlying))
```

Displays:

```
SingleType(NoPrefix, TermName("Person"))
TypeRef(NoPrefix, TypeName("Person"), List())
```

We need to invalidate the caches of the `TypeRef` represending the module class.

I've pushed another commit which deals with this internally in `invalidateCaches`. That could potentially catch other bugs.